src/module/canvas/token/flanking-highlight/renderer.ts

```diff
@@ -0,0 +1,174 @@
+        const buddies = this.token.buddiesFlanking(target, { ignoreFlankable: true });
+        this.drawBuddyLines(buddies);
+    }
+
+    /**
+     * Draw flanking highlight lines between this token and buddies
+     * @param buddies Flanking buddy tokens
+     */
+    drawBuddyLines(buddies: TokenPF2e[]): void {
+        buddies.forEach((b) => this.drawBuddyLine(b));
+    }
+
+    /**
+     * Draw flanking highlight line between this token and a single buddy.
+     * @param buddy Flanking buddy token
+     */
+    drawBuddyLine(buddy: TokenPF2e): void {
+        const t = CONFIG.Canvas.objectBorderThickness;
+        const o = Math.round(t * 1.5);
+        const c = Math.round(t * 2);
+
+        // Draw line
+        this.layer
+            .lineStyle(o, 0x000000, 0.5)
+            .moveTo(this.token.center.x, this.token.center.y)
+            .lineTo(buddy.center.x, buddy.center.y);
+        this.layer
+            .lineStyle(t, this.lineColor, 0.5)
+            .moveTo(this.token.center.x, this.token.center.y)
+            .lineTo(buddy.center.x, buddy.center.y);
+
+        // Draw circles on tokens
+        this.layer
+            .beginFill(this.lineColor)
+            .lineStyle(1, 0x000000)
+            .drawCircle(this.token.center.x, this.token.center.y, c);
+        this.layer.beginFill(this.lineColor).lineStyle(1, 0x000000).drawCircle(buddy.center.x, buddy.center.y, c);
+
+        // Add text indicator
+        this.drawLabel(buddy);
+    }
+
+    /**
+     * Draw a flanking text label above flanking highlight line
+     * @param buddy Flanking buddy token
+     */
+    drawLabel(buddy: TokenPF2e): void {
+        // Midpoint coordinate between tokens
+        const mid_x = Math.round((this.token.center.x + buddy.center.x) / 2);
+        const mid_y = Math.round((this.token.center.y + buddy.center.y) / 2);
+
+        // Vector between tokens
+        const vect_x = buddy.center.x - this.token.center.x;
+        const vect_y = buddy.center.y - this.token.center.y;
+
+        // find the perpendicular vector "above" the line
+        const perp_vect_x = vect_x <= -vect_x ? -vect_y : vect_y;
+        const perp_vect_y = vect_x <= -vect_x ? vect_x : -vect_x;
+
+        // Midpoint coordinate offset perpendicularly above line
+        const offsetPixels = 20.0;
+        const offsetScale = offsetPixels / Math.sqrt(perp_vect_x ** 2 + perp_vect_y ** 2);
+        const perp_x = mid_x + Math.round(perp_vect_x * offsetScale);
+        const perp_y = mid_y + Math.round(perp_vect_y * offsetScale);
+
+        // Styling
+        const style = CONFIG.canvasTextStyle.clone();
+        style.fontSize = 24;
+        if (canvas.dimensions?.size && canvas.dimensions.size >= 200) style.fontSize = 28;
+        else if (canvas.dimensions?.size && canvas.dimensions.size < 50) style.fontSize = 20;
+        style.fill = this.lineColor;
+        style.stroke = 0x000000;
+
+        const text = new PreciseText("Flanking", style);
```

Comment:
Does this need to be localizable?

Reply:
This is done now, this text is localizable.